execution/scan/literal_selector.go

```diff
@@ -39,8 +40,25 @@ func NewNumberLiteralSelector(pool *model.VectorPool, opts *query.Options, val f
 		currentStep: opts.Start.UnixMilli(),
 		val:         val,
 	}
+
+	op.t = &model.NoopTimingInformation{}
+	if opts.EnableAnalysis {
+		op.t = &model.TimingInformation{CPUTime: time.Duration(0)}
+	}
+
+	return op
 }
+func (o *numberLiteralSelector) Analyze() (*model.TimingInformation, []model.ObservableVectorOperator) {
+	if telemetry, ok := o.t.(*model.TimingInformation); ok {
+		_, _ = o.Next(context.Background())
```

Comment:
Ok, I think we are a bit confused here. Do we need to call Next in Analyze? 

Next is only meant to be called for the query execution. Calling Next here would basically mean calling Next twice on each query if analyze param is enabled in https://github.com/thanos-io/thanos/pull/6515/files. Anaylze should only return timing for that operator. The recording of the duration is done just once in Next